runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +162,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TransactionAbortedException) {
+                    TransactionAbortedException tae = (TransactionAbortedException) e;
+                    if (tae.getCause() instanceof TrimmedException) {
+                        TrimmedException te = (TrimmedException) tae.getCause();
+                        AbstractTransactionalContext context = TransactionalContext.getCurrentContext();
+                        System.out.println(te.getSnapshot() + "   " + TransactionalContext
+                                .getCurrentContext().getSnapshotTimestamp());
+                        if (context.getRetry() < 1 || te.getSnapshot() != Address.NON_ADDRESS
```

Comment:
Agree, counter should be for loop only. Also, handling this error is still in the wrong place.